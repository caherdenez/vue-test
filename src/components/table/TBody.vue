<template>
  <tr @click="onClickHandler">
    <td>{{ value.market_cap_rank }}</td>
    <td>{{ value.name }}</td>
    <td><img height="35px" width="35px" :src="value.image" :alt="value.id" /></td>
    <td>{{ value.current_price }}</td>
    <td>{{ value.high_24h }} / {{ value.low_24h }}</td>
  </tr>
</template>

<script>
export default {
  name: 'TBody',
  props: {
    value: {
      type: Object
    }
  },
  methods: {
    onClickHandler() {
      this.$emit('click', this.value);
    }
  }
};
</script>

<style></style>
