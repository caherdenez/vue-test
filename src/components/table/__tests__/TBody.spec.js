import { shallowMount } from '@vue/test-utils';
import TBody from '../TBody.vue';

const data = {
  id: 'bitcoin',
  symbol: 'btc',
  name: 'Bitcoin',
  image: 'https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579',
  current_price: 57956,
  market_cap: 1078125771481,
  market_cap_rank: 1,
  fully_diluted_valuation: 1213482133676,
  total_volume: 61789740666,
  high_24h: 60186,
  low_24h: 57511,
  price_change_24h: 41.5,
  price_change_percentage_24h: 0.07165,
  market_cap_change_24h: -5036211282.281982,
  market_cap_change_percentage_24h: -0.46495,
  circulating_supply: 18657581.0,
  total_supply: 21000000.0,
  max_supply: 21000000.0,
  ath: 61712,
  ath_change_percentage: -6.34474,
  ath_date: '2021-03-13T20:49:26.606Z',
  atl: 67.81,
  atl_change_percentage: 85134.13501,
  atl_date: '2013-07-06T00:00:00.000Z',
  roi: null,
  last_updated: '2021-03-18T21:08:02.135Z'
};

describe('TBody.vue', () => {
  it('render', () => {
    const wrapper = shallowMount(TBody, {
      propsData: {
        value: data
      }
    });
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.vm.onClickHandler();
    const emitData = wrapper.emitted().click[0][0];
    expect(emitData.id).toEqual(data.id);
  });
});
