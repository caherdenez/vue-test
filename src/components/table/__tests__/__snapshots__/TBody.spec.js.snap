// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TBody.vue render 1`] = `
<tr>
  <td>1</td>
  <td>Bitcoin</td>
  <td><img height="35px" width="35px" src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579" alt="bitcoin"></td>
  <td>57956</td>
  <td>60186 / 57511</td>
</tr>
`;
